### PR TITLE
Bug fix/dont use exceptions for control flow

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2017/

File: arangod/Cluster/ServerState.cpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ void ServerState::findHost(std::string const& fallback) {
 
   // Now look at the contents of the file /etc/machine-id, if it exists:
   std::string name = "/etc/machine-id";
-  try {
+  if (arangodb::basics::FileUtils::exists(name)) try {
```
Comment (Simran-B) [inline arangod/Cluster/ServerState.cpp]: This this intended? `if (...) try { ... }`

File: arangod/Cluster/SynchronizeShard.cpp [hunk 2]
```diff
@@ -688,9 +686,8 @@ bool SynchronizeShard::first() {
       _result.reset(TRI_ERROR_FAILED, msg.str());
```
Comment (ObiWahn) [inline arangod/Cluster/SynchronizeShard.cpp]: Could we copy the result of `ci` here to get a more exact error?

File: arangod/ClusterEngine/RocksDBMethods.cpp [hunk 3]
```diff
@@ -52,21 +52,17 @@ Result recalculateCountsOnAllDBServers(std::string const& dbname,
   }
   
   // First determine the collection ID from the name:
-  std::shared_ptr<LogicalCollection> collinfo;
-  try {
-    collinfo = ci->getCollection(dbname, collname);
-  } catch (...) {
+  auto collinfo = ci->getCollection(dbname, collname);
+  if (collinfo.fail()) {
     return TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND;
```
Comment (ObiWahn) [inline arangod/ClusterEngine/RocksDBMethods.cpp]: Use error and message of failed `collinfo`?

**Comment by graetzer** [comment]
Really stupid question: why do you not just return nullptr (or `std::shared_ptr<LogicalCollection>(nullptr)`) in `ClusterInfo::getCollection` if the collection is not found? 
We do not really need the formatted error message that deep in the API or do we ?

**Comment by dothebart** [comment]
superseeded by https://github.com/arangodb/arangodb/pull/7339

**Pull request closed without merging**